filament/src/UniformBuffer.h

```diff
@@ -123,8 +124,15 @@ class UniformBuffer { // NOLINT(cppcoreguidelines-pro-type-member-init)
     // offset in bytes, and count is the number of elements to invalidate
     template <typename T, typename = typename is_supported_type<T>::type>
     void setUniformArray(size_t offset, T const* UTILS_RESTRICT begin, size_t count) noexcept {
-        T* UTILS_RESTRICT p = static_cast<T*>(invalidateUniforms(offset, sizeof(T) * count));
-        std::copy_n(begin, count, p);
+        // we need to align array elements to the size of a vec4 (see std140 layout)
+        constexpr size_t stride = (sizeof(T) + 0xF) & ~0xF;
+        T* UTILS_RESTRICT p = static_cast<T*>(invalidateUniforms(offset, stride * count));
+        for (size_t i = 0; i < count; i++) {
+            *p++ = begin[i];
+            if (sizeof(T) & 0xF) {
```

Comment:
Make sure this is optimized with float4, and should we remove the float3 specialization.